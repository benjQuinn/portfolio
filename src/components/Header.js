import React from "react";
import "../styles/header.css";

function Header() {
  return (
    <>
      <div className="header">
        <h1 className="header-first-name">B</h1>
        <h1 className="header-first-name">E</h1>
        <h1 className="header-first-name">N</h1>
        <h1 className="header-last-name">Q&nbsp;U&nbsp;I&nbsp;N&nbsp;N</h1>
      </div>
      <h2 className="occupation">FULL-STACK SOFTWARE DEVELOPER</h2>
    </>
  );
}

export default Header;
