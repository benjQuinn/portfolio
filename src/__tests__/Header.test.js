import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Header from "../components/Header";

describe("Header", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders text", () => {
    render(<Header />);

    expect(screen.getByText("B")).toBeInTheDocument();
    expect(screen.getByText("E")).toBeInTheDocument();
    expect(screen.getByText("N")).toBeInTheDocument();
    expect(screen.getByText(/q/i)).toBeInTheDocument();
    expect(
      screen.getByText(/full-stack software developer/i)
    ).toBeInTheDocument();
  });
});
