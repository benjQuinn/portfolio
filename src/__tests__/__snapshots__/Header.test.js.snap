// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <h1
      class="header-first-name"
    >
      B
    </h1>
    <h1
      class="header-first-name"
    >
      E
    </h1>
    <h1
      class="header-first-name"
    >
      N
    </h1>
    <h1
      class="header-last-name"
    >
      Q U I N N
    </h1>
  </div>
  <h2
    class="occupation"
  >
    FULL-STACK SOFTWARE DEVELOPER
  </h2>
</DocumentFragment>
`;
